mk/compile.mk

```diff
@@ -17,8 +17,8 @@ objs		:=
 # Disable all builtin rules
 .SUFFIXES:
 
-__cc-option = $(if $(shell $(CC$(sm)) $(1) -c -x c /dev/null -o /dev/null 2>&1 >/dev/null),$(2),$(1))
-_cc-opt-cached-var-name = cached-cc-option$(subst =,~,$(strip $(1)))$(subst $(empty) $(empty),,$(CC$(sm)))
+__cc-option = $(if $(shell $(CC$(sm)) $(1) -c -x c /dev/null -o /dev/null 2>/dev/null >/dev/null || echo "Not supported"),$(2),$(1))
+_cc-opt-cached-var-name = $(subst =,~,$(strip cached-cc-option-$(1)-$(subst $(empty) $(empty),,$(CC$(sm)))))
```

Comment:
i don't catch this change :)  LGTM AFAICT.

Reply:
I admit it is not super easy to read :-/ but that is the case for many gmake macros anyway... I am simply applying the `subst()` to the whole string (and adding two more dashes `-` but that's cosmetic).

More precisely... Here we are creating a name for a variable used to cache the detected option (we don't want to call the compiler again and again). This name must contain the compiler command and the option we are testing.
What this does is (1) remove spaces from the compiler command and (2) replace all equal signs `=` with a tilde `~` in the whole name. Previously we were replacing only in the option, not in the compiler command.
The option can contain `=`, for instance `-mcpu=cortex-a53`. The compiler command can also contain an equal sign with Clang: `clang --target=arm-linux-gnueabihf`.

For example, say we want to check if option `-mcpu=cortex-a53` is supported by Clang during a 32-bit build. This can be done with: `$(call cc-option,-mcpu=cortex-a53,)`. Then:
- `$(CC$(sm))` is `clang --target=arm-linux-gnueabihf`
- `$(1)` is `-mcpu=cortex-a53`
- `$(_cc-opt-cached-var-name)` is `cached-cc-option--mcpu~cortex-a53-clang--target~arm-linux-gnueabihf`